src/netbase.cpp

```diff
@@ -651,7 +667,7 @@ void CNetAddr::SetRaw(Network network, const uint8_t *ip_in)
             memcpy(ip, ip_in, 16);
             break;
         default:
-            assert(!"invalid network");
+	    assert(true);  // invalid network
```

Comment:
I changed the code to eliminate the weirdness and subsequent diagnostic.  Other than that, I tried not to change the behavior which was to only abort during debug.

I'm not familiar with a version of assert that prints the argument.  Here is the definition:

If NDEBUG is not defined, then assert checks if its argument (which must have scalar type) compares equal to zero. If it does, assert outputs implementation-specific diagnostic information on the standard error output and calls std::abort.